### PR TITLE
[DROOLS-7107] Drop or Deprecate drools-templates

### PR DESCRIPTION
**Pull request by tkobayas**
**JIRA**: 

https://issues.redhat.com/browse/DROOLS-7107

https://github.com/kiegroup/drools/pull/4611#discussion_r949208103
@mariofusco mentioned
> I agree that our custom made rule templates implementation is brittle and not well maintained and that we could rewrite it using a more standard and battle tested templating engine. My suggestion is taking advantage of the new major release removing not only this paragraph but also the implementation code (or at least declaring it deprecated).

Actually `drools-decisiontables` depends on `drools-templates` so we cannot remove `drools-templates`. So now I raise a warning when a user uses `DRT` file. Also added a warning in the document. Is it good enough?

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] tests</b>

- for a <b>full downstream build</b> 
  - for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  - for <b>github actions</b> job: add the label `run_fdb`

- <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

- <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

- <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] quarkus-branch</b>

- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] native</b>

- for <b>mandrel checks</b>  
  Run native checks against Mandrel image
  Please add comment: <b>Jenkins run mandrel</b>

- for a <b>specific mandrel check</b>  
  Run native checks against Mandrel image  
  Please add comment: <b>Jenkins (re)run [drools|kogito-runtimes|kogito-apps|kogito-examples] mandrel</b>
</details>

### CODE REVIEW

**Comment by kie-ci3** [comment]
**(tests) - drools job** `#503` was: **UNSTABLE**
Possible explanation: This should be test failures

Please look here: https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.drools/503/display/redirect

**Test results:**
- PASSED: 20585
- FAILED: 2

Those are the test failures: 
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.drools/503/testReport/org.kie.dmn.feel.client.showcase/FEELShowcaseIT/Build_projects___testEvaluation/">Build projects / org.kie.dmn.feel.client.showcase.FEELShowcaseIT.testEvaluation</a></summary>
Process unexpectedly closed with status 0<br/>Build info: version: '3.13.0', revision: '2f0d292', time: '2018-06-25T15:24:21.231Z'<br/>System info: host: 'eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com', ip: '127.0.0.1', os.name: 'Linux', os.arch: 'amd64', os.version: '3.10.0-1160.76.1.el7.x86_64', java.version: '11.0.16'<br/>Driver info: driver.version: FirefoxDriver<br/>remote stacktrace: 
</details>
<details>
<summary><a href="https://eng-jenkins-csb-business-automation.apps.ocp-c1.prod.psi.redhat.com/job/KIE/job/kogito/job/main/job/pullrequest/job/drools.tests.drools/503/testReport/org.kie.dmn.feel.client.showcase/FEELShowcaseIT/Build_projects___testEvaluation/">Build projects / org.kie.dmn.feel.client.showcase.FEELShowcaseIT.testEvaluation</a></summary>
java.lang.NullPointerException<br/>	at org.kie.dmn.feel.client.showcase.FEELShowcaseIT.quitDriver(FEELShowcaseIT.java:61)<br/>
</details>

File: kie-api/src/main/java/org/kie/api/io/ResourceType.java [hunk 1]
```diff
@@ -251,6 +259,9 @@ public static ResourceType addResourceTypeToRegistry(final String resourceType,
     public static ResourceType determineResourceType(final String resourceName) {
         for ( Map.Entry<String, ResourceType> entry : CACHE.entrySet() ) {
             if (resourceName.endsWith(entry.getKey())) {
+                if (entry.getValue().equals(ResourceType.DRT)) {
+                    LOG.warn("DRT (Drools Rule Template) is deprecated. Please consider drools-decisiontables or third party templating features.");
```
Comment (danielezonca) [inline kie-api/src/main/java/org/kie/api/io/ResourceType.java:263]: Given that decisiontables use templates, is this warning raised also using decision table?

**Review by mariofusco — APPROVED** [review]

File: kie-api/src/main/java/org/kie/api/io/ResourceType.java (hunk 1 above)
Comment (tkobayas) [reply kie-api/src/main/java/org/kie/api/io/ResourceType.java:263]: No. decisiontables uses XLS or XLSX file. The warning is raised only when you use DRT file (suffix).  `drools-decisiontables` depends on `drools-templates` in more internal ways.

**Comment by tkobayas** [comment]
SonarCloud fails with a Minor code smell `Remove this use of "DRT"; it is deprecated.` but I needed to use it to raise a warning "it's deprecated" :D

I think we can ignore the failure.

**Comment by tkobayas** [comment]
@mariofusco Could you merge this PR? Thanks!

File: kie-api/src/main/java/org/kie/api/io/ResourceType.java (hunk 1 above)
Comment (danielezonca) [reply kie-api/src/main/java/org/kie/api/io/ResourceType.java:263]: Perfect, thanks for the confirmation 👍

**Pull request merged**